### PR TITLE
Adjust keras to use Constant for vars. Fix closure related bugs

### PR DESCRIPTION
**Pull request by flaub**

### CODE REVIEW

**Review by shiyao00 — DISMISSED** [review]
LGTM

File: pmlc/dialect/stdx/tests/split_closure.mlir [hunk 1]
```diff
@@ -58,7 +58,7 @@ func @main(%arg0: memref<16x16xf32> {stdx.const = 0 : index}, %arg1: memref<16x1
 
 // -----
 
-func @main(%arg0: memref<16x16xf32> {stdx.const = 0 : index}, %arg1: memref<16x16xf32> {stdx.const = 1 : index}) {
+func @main(%arg0: memref<16x16xf32> {stdx.conststdx.const}, %arg1: memref<16x16xf32> {stdx.const}) {
```
Comment (tzerrell) [inline pmlc/dialect/stdx/tests/split_closure.mlir]: Typo: `stdx.conststdx.const`

**Review by tzerrell — COMMENTED** [review]
LGTM. One inline typo.

File: pmlc/dialect/stdx/tests/split_closure.mlir (hunk 1 above)
Comment (flaub) [reply pmlc/dialect/stdx/tests/split_closure.mlir]: Good catch

**Review by tzerrell — APPROVED** [review]

**Pull request merged**